realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -1025,6 +1025,39 @@ private Table getPrimaryKeyTable() {
         return pkTable;
     }
 
+    /**
+     * Rename the primary key column for the table. This needs to be called manually before inserting data into the table.
+     *
+     * @param newFieldName the name of the field replacing old primary key field name.
+     * @throws {@link IllegalArgumentException} if {@code newFieldName} is an empty string, or exceeds field name length limit.
+     * @throws {@link io.realm.exceptions.RealmException} if a primary key does not exist.
+     */
+    public void renamePrimaryKeyColumn(String newFieldName) {
+        if (newFieldName == null || newFieldName.length() == 0) {
+            throw new IllegalArgumentException("Primary key field cannot be renamed with an empty string.");
+        }
+        verifyColumnName(newFieldName);
+        long pkColumnIndex = getPrimaryKey();
+        if (pkColumnIndex == NO_PRIMARY_KEY) {
+            throw new RealmException("Non-existent primary key field cannot be renamed.");
```

Comment:
I would fold this method into `renameColumn`. That would make this exception not needed.